jitsi_uploader.sh

```diff
@@ -1,143 +1,176 @@
 #!/bin/bash
 
-CONFIG_FILE_PATH="/etc/jitsi/uploader"
+[ -z "$CONFIG_FILE_PATH" ] && CONFIG_FILE_PATH="/etc/jitsi/uploader"
 
 # pull jitsi uploader environment var overrides if it exists
 [ -e  "$CONFIG_FILE_PATH" ] && . "$CONFIG_FILE_PATH"
 
 #make this backed by a decent sized disk
 [ -z "$FAILED_UPLOAD_DIR" ] && FAILED_UPLOAD_DIR="/tmp/failed"
 
+
 #assume supporting binaries are in /usr/bin
 BIN_PATH="/usr/bin"
 
-UPLOAD_DIR=$1
+#if not provided, path for custom recording script is pre-defined 
+[ -z "$CUSTOM_RECORDING_SERVICE_PATH" ] && CUSTOM_RECORDING_SERVICE_PATH="$BIN_PATH/jitsi-recording-service.sh"
 
-if [ -z "$UPLOAD_DIR" ]; then
-    echo "No upload directory provided, failing..."
-    exit 1
-fi
+#db uploader function
+# $1 - current path to source file
+# $2 - full path of destination
+function dropbox_upload {
+    echo "Upload method: dropbox"
+    UPLOAD_BIN="$BIN_PATH/dropbox_uploader.sh"
+    export OAUTH_ACCESS_TOKEN=$TOKEN
+    $UPLOAD_BIN -b upload "$1" "$2"
+    return $?
+}
 
-if [ ! -d "$UPLOAD_DIR" ]; then
-    echo "No such directory $UPLOAD_DIR, failing..."
-    exit 2
-fi
+function preprocess_upload_dir {
 
-METADATA_JSON="$UPLOAD_DIR/metadata.json"
+  UPLOAD_DIR=$1
 
-if [[ ! -e "$METADATA_JSON" ]]; then
-  echo "No file found $METADATA_JSON, failing."
-  exit 3
-fi
+  if [ -z "$UPLOAD_DIR" ]; then
+      echo $(date) "No upload directory provided, failing..."
+      return 1
+  fi
 
-#only read token from metadata if token is not already defined
-[[ -z "$TOKEN" ]] && TOKEN=$(cat $METADATA_JSON | jq -r ".upload_credentials.token")
-[[ "$TOKEN" == "null" ]] && TOKEN=""
-[[ -z "$UPLOAD_TYPE" ]] && UPLOAD_TYPE=$(cat $METADATA_JSON | jq -r ".upload_credentials.service_name")
-[[ "$UPLOAD_TYPE" == "null" ]] && UPLOAD_TYPE=""
-URL=$(cat $METADATA_JSON | jq -r ".meeting_url")
-[[ "$URL" == "null" ]] && URL=""
-URL_NAME="${URL##*/}"
+  if [ ! -d "$UPLOAD_DIR" ]; then
+      echo $(date) "No such directory $UPLOAD_DIR, failing..."
+      return 2
+  fi
 
-if [[ -z "$UPLOAD_TYPE" ]]; then
+  METADATA_JSON="$UPLOAD_DIR/metadata.json"
 
-  if [[ -f $BIN_PATH/jitsi-recording-service.sh ]] && [[ -x $BIN_PATH/jitsi-recording-service.sh ]]; then
-    $BIN_PATH/jitsi-recording-service.sh $UPLOAD_DIR
-    exit $?
+  if [[ ! -e "$METADATA_JSON" ]]; then
+    echo $(date) "No file found $METADATA_JSON, failing."
+    return 3
   fi
 
-  echo "No upload type found, skipping upload..."
-  exit 4
-fi
-
+  #only read token from metadata if token is not already defined
+  [[ -z "$TOKEN" ]] && TOKEN=$(cat $METADATA_JSON | jq -r ".upload_credentials.token")
```

Comment:
could the token come from elsewhere?

Reply:
Yes, it could come in through the TOKEN environment variable.  This was mostly kept around from previous iterations where a token was generated outside the uploader and passed in via environment var.